### DISCUSSION TITLE
HttpResponseCode: 500 - No retry

### DISCUSSION THREAD

**Discussion by justcode0012** [pnp/pnpcore#1305] [Q&A] [answered]
Hello,

in the past i got an HTTP 500 from SharePoint.
After investigating the code (RetryHandlerBase -> ShouldRetry) i recognized that in this case a retry does not occur.
Is there any specific reason for that? In my case (simple update of a SharePoint ListItem) a retry should have resolved this issue. It seems that is was a temporary issue.

Logged error message:
PnP.Core.Services.BatchClient+<ExecuteSharePointRestInteractiveAsync>d__60 Runtime exception encounteredHttpResponseCode: 500
Code: Microsoft.SharePoint.SPException
Message: Exception from HRESULT: 0x80131904
ClientRequestId: 4c5ce5a0-00d4-7000-68ad-1c42598d2ef5
SPClientServiceRequestDuration: 352
X-SharePointHealthScore: 0
X-SP-SERVERSTATE: ReadOnly=0

PnP.Core.SharePointRestServiceException: SharePoint Rest service exception
   at PnP.Core.Services.BatchClient.ExecuteSharePointRestInteractiveAsync(Batch batch)
   at PnP.Core.Services.BatchClient.ExecuteSharePointRestBatchAsync(Batch batch)
   at PnP.Core.Services.BatchClient.ExecuteBatch(Batch batch)
   at PnP.Core.Model.SharePoint.ListItem.BaseUpdate(Func`2 fromJsonCasting, Action`1 postMappingJson)
   at PnP.Core.Model.BaseDataModel`1.UpdateAsync()

Best Regards
Dennis

**Comment by jansenbe** [comment, marked as answer]
@dennisvanniel : retrying on errors is not something we can do automatically as for most errors they're not transient and you'll just get the same error response on each try. The way forward is for you to catch this in custom code on the places where you feel it makes sense.
